src/python/pants/testutil/rule_runner.py

```diff
@@ -476,7 +476,10 @@ def get_target(self, address: Address) -> Target:
 
         :API: public
         """
-        return self.request(WrappedTarget, [address]).target
+        return self.request(
+            WrappedTarget,
+            [WrappedTargetRequest(address, description_of_origin="RuleRunner.get_target()")],
```

Comment:
I love this one :)

Reply:
Hehe, nice. And I see avoidable boilerplate. The engine already knows which rule the `await Get()` comes from, so the rule graph could have some sort of execution context, that each type have an associated origin rule attached to it. (which when used for error messages, the rule description, or something else, could be extracted from that context)

Well, actually in this particular case, the above would not be true, as this is not executed from within a `@rule` body. But the same logistics could still be applied, only that the `request()` method would need to introspect the callstack to find the point where the call was made, and point at that.